packages/fast-form-generator-react/src/form/form.utilities.ts

```diff
@@ -218,6 +227,156 @@ export function getComponentTracker(
     return updatedComponentTracker;
 }
 
+/**
+ * Creates a schema location from a data location
+ */
+export function mapSchemaLocationFromDataLocation(dataLocation: string, data: any, schema: any): string {
+    const squareBracketRegex: RegExp = /\[(\d+?)\]/g;
+    let normalizedDataLocation: string = dataLocation.replace(squareBracketRegex, `.$1`); // convert all [ ] to . notation
+    normalizedDataLocation = convertArrayItemsToBracketNotation(dataLocation, data); // convert back all array items to use [ ]
+    const dataLocationStrings: string[] = normalizedDataLocation.split(".");
+
+    if (dataLocation !== "") {
+        dataLocationStrings.unshift("");
+    }
+
+    const schemaLocationStrings: string[] = getSchemaLocationStringsFromDataLocationStrings(dataLocationStrings, schema, data);
+
+    return schemaLocationStrings.join(".").replace(squareBracketRegex, "");
+}
+
+/**
+ * Get an array of schema location strings from an array of data location strings
+ */
+export function getSchemaLocationStringsFromDataLocationStrings(dataLocationStrings: string[], schema: any, data: any): string[] {
+    const squareBracketRegex: RegExp = /\[(\d+?)\]/g;
+    let schemaLocationStrings: string[] = [];
+    let reconstitutedDataLocation: string = "";
+
+    for (let i: number = 0; i < dataLocationStrings.length; i++) {
+        const partialData: any = getPartialData(reconstitutedDataLocation, data);
```

Comment:
For both calls to `getPartialData` you're putting together location segments - it seems you could simplify by doing the work to convert arguments to dot notation inside of the `getPartialData` function instead. This then turns into:

```
const partialData: any = getPartialData(dataLocationStrings.slice(0, index), data);
const partialSchema: any = getPartialData(schemaLocationStrings, schema);
```

You can then remove `reconstitutedDataLocation` all together.